src/libsyntax/attr.rs

```diff
@@ -205,6 +211,10 @@ impl NestedMetaItem {
     }
 }
 
+fn name_from_path(path: &ast::Path) -> Name {
+    path.segments.iter().next().unwrap().identifier.name
```

Comment:
Could you use the last segment instead?
In `#[a::b::c]` `a` and `b` are modules (possibly "tool modules") in which the attribute `c` is defined, and the last segment `c` is the attribute itself.